docs/python/doxygenlib/cdWriterDocstring.py

```diff
@@ -295,11 +272,14 @@ def generate(self, output_file, docElements):
         if len(lines) == 1:
             lines.append("    pass")
 
+        outputDir = os.path.split(output_file)[0]
+        if not os.path.exists(outputDir):
+            os.makedirs(outputDir)
+
         # output the lines to disk...
         try:
-            logfile = open(output_file, 'w')
-            logfile.write('\n'.join(lines))
-            logfile.close()
+            with open(output_file, 'w') as logfile:
+                logfile.write('\n'.join(lines))
```

Comment:
use context manager to ensure that the file is closed in case of an exception (also, more idiomatic)